src/sentry/shared_integrations/exceptions/__init__.py

```diff
@@ -57,6 +58,10 @@ class ApiRateLimitedError(ApiError):
     code = 429
 
 
+class ApiConnectionResetError(ApiError):
+    code = 104
```

Comment:
Hmmm... Now I'm wondering, but first, here's how I got there:

- In the live examples we have, it's `ConnectionResetError(104, "Connection reset by peer")`
- If you look up `ConnectionResetError` in [the Python docs](https://docs.python.org/3/library/exceptions.html#ConnectionResetError), it's listed as a subclass of `ConnectionError`, which is itself a subclass of `OSError`.
- According to [the docs](https://docs.python.org/3/library/exceptions.html#OSError) for `OSError`, it's first argument is an error code. Hence, the `104` we have is presumably the error code for a `ConnectionResetError`.

Now I'm puzzled, though, because I went and figured out the actual value for [`ECONNRESET`](https://docs.python.org/3/library/errno.html#errno.ECONNRESET) (the constant associated with `ConnectionResetError`) by printing `errno.errorcode` ([the dictionary](https://docs.python.org/3/library/errno.html#errno.errorcode) mapping codes to error constants) in my local Python and got... 54. Maybe it's platform-dependent? (EDIT: Yes, seems [it is](https://stackoverflow.com/questions/961465/identical-error-codes).)

Given this mismatch, I'm going to switch to using the constant. Technically it's not guaranteed to be a particular value in all cases this code could be run ever, but the chances of someone running their sentry server in linux one day and in OSX another are IMHO pretty small, so for any given instance of Sentry, it should be stable.